dynawo/sources/Models/CPP/ModelNetwork/DYNModelCurrentLimits.cpp

```diff
@@ -57,60 +58,42 @@ ModelCurrentLimits::setMaxTimeOperation(const double& maxTimeOperation) {
   maxTimeOperation_ = maxTimeOperation;
 }
 
-void
-ModelCurrentLimits::setNbLimits(const int& number) {
-  nbLimits_ = number;
-  activated_.assign(number, false);
-  tLimitReached_.assign(number, std::numeric_limits<double>::quiet_NaN());
-}
-
 void
 ModelCurrentLimits::addLimit(const double& limit) {
-  limits_.push_back(limit);
-  if (std::isnan(limit))
-    limitActivated_.push_back(false);
-  else
-    limitActivated_.push_back(true);
+  if (!std::isnan(limit)) {
+    limits_.push_back(limit);
+    activated_.push_back(false);
+    tLimitReached_.push_back(std::numeric_limits<double>::quiet_NaN());
+    nbLimits_++;
+  }
 }
 
 void
 ModelCurrentLimits::addAcceptableDuration(const int& acceptableDuration) {
-  acceptableDurations_.push_back(acceptableDuration);
-  if (acceptableDuration == std::numeric_limits<int>::max())
+  if (acceptableDuration == std::numeric_limits<int>::max()) {
     openingAuthorized_.push_back(false);
-  else
+  } else {
     openingAuthorized_.push_back(true);
+    acceptableDurations_.push_back(acceptableDuration);
```

Comment:
with this there is a mismatch between acceptableDurations_.size() and nbLimits.size() and thus we lose the information of which Limits correspond to which duration